src/components/CreateItem.jsx

```diff
@@ -9,19 +10,37 @@ function CreateItem() {
     frequency: '7',
   };
   const [item, setItem] = useState(itemObj);
+  const token = localStorage.getItem('token');
+
+  const [value, loading, error] = useCollection(
+    db.collection('items').where('token', '==', token),
+  );
 
   const handleSubmit = (e) => {
     e.preventDefault();
-    const token = localStorage.getItem('token');
 
-    db.collection('items').add({
-      name: item.itemName,
-      frequency: item.frequency,
-      lastPurchasedDate: null,
-      date: Date().toLocaleString(),
-      token,
-    });
-    setItem(itemObj);
+    const findDuplicateItem = (item) => {
+      let existingItem = false;
+      value.docs.map((doc) => {
+        if (doc.data().name === item.itemName) {
+          existingItem = true;
+        }
+      });
+      return existingItem;
+    };
+
+    const duplicatedItem = findDuplicateItem(item);
+
+    if (!duplicatedItem) {
+      db.collection('items').add({
+        name: item.itemName,
+        frequency: item.frequency,
+        lastPurchasedDate: null,
+        date: Date().toLocaleString(),
+        token,
+      });
+      setItem(itemObj);
+    } else alert('The item already exist');
```

Comment:
This fulfills the AC of showing the user an error instead of adding the new item, but maybe as we go forward and make the app more styled and user friendly it might be worth thinking of showing it in a more accessible way than an alert pop-up? 🤔 I'm not sure what the right answer is here, and obviously alert() does the job, but I think it might be worth thinking about in future weeks ☺️

Reply:
There are many different ways to do it, and I'd recommend moving away from the alert, it's a bit aggressive. One way to do it is to pull in [another library](https://www.cssscript.com/best-toast-notification-libraries/), there are tons of them.

Another way would be to just add an error element when something like that occurs.

Reply:
Yup I agree! maybe when it's time for styling you can implement a more accessible alert or a modal too.👍👍

Reply:
These are all great suggestions. One more idea I could propose is to maybe coordinate with the other pair and implement their notification message logic.

One thing you can do it so maybe create a new component that holds the notification message logic and re-use it in other components. Let me know if it makes sense 🙌